### PR TITLE
Add an option bundle to the loader, fix square view atop the goal picker

### PR DESCRIPTION
**Pull request by Revenaunt**
Now, calls to loadBitmap need to provide an option bundle with the parameters of the load.

### CODE REVIEW

File: src/main/java/org/tndata/android/compass/util/ImageLoader.java [hunk 1]
```diff
@@ -707,4 +664,69 @@ protected void onPostExecute(Void result){
             }
         }
     }
+
+    private static void completeLodRequest(LoadRequest request){
```
Comment (bradmontgomery) [inline src/main/java/org/tndata/android/compass/util/ImageLoader.java]: Typo?   Should this be `completeLoadRequest` ?

Comment (Revenaunt) [inline src/main/java/org/tndata/android/compass/util/ImageLoader.java]: Totally typo. One sec.

File: src/main/java/org/tndata/android/compass/util/ImageLoader.java [hunk 2]
```diff
@@ -679,14 +636,14 @@ else if (cancelPotentialWork(request.mUrl, request.mImageView)){
...
+        public void onDownloadComplete(LoadRequest request, boolean wasCancelled){
+            Log.d("ImageLoader", "Download complete: " + request.mUrl);
             //When a download is completed, if the task wasn't cancelled the content is queued
             //  to be written to cache. If the task was cancelled, the Bitmap might be corrupt.
-            if (!wasCancelled && result != null){
-                MemoryCache.instance().addBitmapToMemoryCache(url, result);
-                queueWriteRequest(new WriteRequest(result, url));
+            if (!wasCancelled && request.mResult != null){
+                MemoryCache.instance().addBitmapToMemoryCache(request.mUrl, request.mResult);
+                completeLodRequest(request);
```
Comment (bradmontgomery) [inline src/main/java/org/tndata/android/compass/util/ImageLoader.java]: Same here... `completeLoadRequest(request)` ?

**Comment by bradmontgomery** [comment]
:+1:

**Pull request merged**